### PR TITLE
feat(all): rc2 updates

### LINKED ISSUE #691: RC2 todo
- [x] 1. https://github.com/neo-project/neo/commit/8fe3dbb962c78979b3f2de85ce1499c1510493a2 (breaking all addresses again 😡 )
- [x] 2. rename `DesignationContract` to `RoleManagement` https://github.com/CityOfZion/neon-js/pull/690#discussion_r598056845
- [x] 3. ~~https://github.com/neo-project/neo/pull/2407~~ (doesn't actually affect neon-js)
- [x] 4. https://github.com/neo-project/neo/pull/2440
- [x] 5. https://github.com/neo-project/neo/pull/2435
- [x] 6. https://github.com/neo-project/neo/pull/2420
- [ ] 7. Update release documentation

### PR DESCRIPTION
**Pull request by ixje**
close #691

### CODE REVIEW

File: packages/neon-api/src/NetworkFacade.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export class NetworkFacade {
 
   async initialize(): Promise<void> {
     const response = await this.client.getVersion();
-    this.magicNumber = response.magic;
+    this.magicNumber = response.network;
```
Comment (ixje) [inline packages/neon-api/src/NetworkFacade.ts:48]: There are still a couple of places left in the code where there are `magicNumber` or `networkMagic`. On the C# side they renamed everything to `network`. You might want to think about if the remaining places are worth renaming or not.

File: packages/neon-core/src/sc/util.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export function isSignatureContract(input: HexString): boolean {
     script[0] != OpCode.PUSHDATA1 ||
     script[1] != PUBLIC_KEY_LENGTH ||
     script[35] != OpCode.SYSCALL ||
-    script.readUInt32LE(36) != 2859889780
+    script.readUInt32LE(36) != 666101590
```
Comment (snowypowers) [inline packages/neon-core/src/sc/util.ts:19]: Can we take this opportunity to derive this magic number from the InteropServiceCode reference so it is clearer?

File: packages/neon-core/src/sc/util.ts [hunk 3]
```diff
@@ -89,7 +89,7 @@ export function isMultisigContract(input: HexString): boolean {
 
   i += 2;
 
-  if (script.readUInt32LE(i) != 2775371387) {
+  if (script.readUInt32LE(i) != 987549854) {
```
Comment (snowypowers) [inline packages/neon-core/src/sc/util.ts:92]: same with this magic number.

**Review by snowypowers — APPROVED** [review]

**Comment by ixje** [comment]
I’ll have a look at it tomorrow

**Pull request merged**